extensions/src/ACRE2Arma/common/dispatch.hpp

```diff
@@ -137,19 +137,19 @@ namespace acre {
                             }
 #endif
                         std::string result_message;
-                        std::int32_t result_code = -1;
-                        dispatcher::call(_message.command, _message.args, result_message, result_code);
+                        std::int32_t result_dummy = 0;
+                        dispatcher::call(_message.command, _message.args, result_message, result_dummy);
                         std::stringstream ss;
                         ss << "[" << _message.id << ",[" << result_message << "]]";
 
                         bool callback_recieved = false; // call back buffer can only hold 100 messages a frame (doubt acre will ever hit this limit)
                         while (!_stop && !callback_recieved) {
-                            int callback_buffer_remaining = callbackFunc("ACRE_TR", _message.command.c_str(), ss.str().c_str());
+                            std::int32_t callback_buffer_remaining = callbackFunc("ACRE_TR", _message.command.c_str(), ss.str().c_str());
                             // LOG(TRACE) << "sending callback [id: " << _message.id << ", buffer: " << callback_buffer_remaining << "]";
                             if (callback_buffer_remaining > -1) {
                                 callback_recieved = true;
                             } else {
-                                sleep(5);
+                                std::this_thread::sleep_for(std::chrono::milliseconds(5));
```

Comment:
The original call `sleep(5)` stops the process for 5 seconds.
```suggestion
                                std::this_thread::sleep_for(std::chrono::seconds(5));
```

Reply:
5 ms is what I wanted originally
extensions push messages the the callback queue, which only has a max depth of 100
and then every in-game-frame it gets emptied
so all this is trying to do is wait for the next in-game frame

Reply:
Ok. Then it is fine